contracts/exchange/mixins/DcvSale.sol

```diff
@@ -0,0 +1,243 @@
+
+        _finalizeSale(
+            sale.nftContract,
+            sale.tokenId,
+            sale.seller,
+            sale.price,
+            royaltyRecipient,
+            platformFee,
+            royaltyFee
+        );
+    }
+
+    function buyFromSaleWithRoyalty(
+        bytes calldata sellerSignature,
+        bytes calldata royaltySignature,
+        SaleWithRoyalty calldata sale
+    ) external payable nonReentrant {
+        if (sale.seller == msg.sender) revert InvalidCaller();
+        if (sale.seller != sale.royaltyRecipient) {
+            revert UnauthorizedRoyaltyChange();
+        }
+
+        _verifySaleWithRoyaltySignature(
+            sellerSignature,
+            royaltySignature,
+            sale
+        );
+
+        _requireSufficientEtherSupplied(sale.price);
+
+        _invalidateNonce(sale.seller, sale.orderNonce);
+
+        uint256 platformFee = _calculateFeeAndSetRoyalty(
+            sale.nftContract,
+            sale.tokenId,
+            sale.price,
+            sale.royaltyRecipient,
+            sale.royaltyPercentage
+        );
+
+        _finalizeSale(
+            sale.nftContract,
+            sale.tokenId,
+            sale.seller,
+            sale.price,
+            sale.royaltyRecipient,
+            platformFee,
+            0
+        );
+    }
+
+    function _finalizeSale(
+        address nftContract,
+        uint256 tokenId,
+        address seller,
+        uint256 price,
+        address royaltyRecipient,
+        uint256 platformFee,
+        uint256 royaltyFee
+    ) internal {
+        IERC721(nftContract).transferFrom(seller, msg.sender, tokenId);
+
+        _distributeEtherFunds(
+            price,
+            royaltyFee,
+            royaltyRecipient,
+            platformFee,
+            seller
+        );
+
+        emit SaleSuccess(
+            nftContract,
+            tokenId,
+            seller,
+            msg.sender,
+            price,
+            platformFee
+        );
+    }
+
+    function _verifySaleSignature(
+        bytes calldata signature,
+        Sale calldata sale
+    ) internal view {
+        bytes32 digest = keccak256(
+            abi.encodePacked(
+                "\x19\x01",
+                _buildDomainSeparator(),
+                keccak256(
+                    abi.encode(
+                        SALE_TYPEHASH,
+                        sale.orderNonce,
+                        sale.nftContract,
+                        sale.tokenId,
+                        sale.price
+                    )
+                )
+            )
+        );
+
+        if (digest.recover(signature) != sale.seller) {
+            revert InvalidSignature();
+        }
+    }
+
+    function _verifySaleWithRoyaltySignature(
+        bytes calldata sellerSignature,
+        bytes calldata royaltySignature,
+        SaleWithRoyalty calldata sale
+    ) internal view {
+        _verifyRoyaltyParameters(royaltySignature, sale);
```

Comment:
Sta kazes da ovo zovemo na istom levelu kao i _verifySaleWithRoyaltySignature? 

Nekako je skriveno u ovoj metodi pa nije jasno na prvu gdje se verificira ovaj potpis?

Reply:
Moze, to sam se bia dvoumia kako cu stavit da bude bolja citljivost